file/types.go

```diff
@@ -444,24 +444,27 @@ func (p *FPlugin) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
-// id is used for sorting.
-func (p FPlugin) id() string {
-	if p.ID != nil {
-		return *p.ID
-	}
+// sortKey is used for sorting.
+func (p FPlugin) sortKey() string {
 	// concat plugin name and foreign relations
-	key := ""
-	key = *p.Name
-	if p.Consumer != nil {
-		key += *p.Consumer.ID
-	}
-	if p.Route != nil {
-		key += *p.Route.ID
+	if p.Name != nil {
```

Comment:
non-blocker: you could simplify this. A plugin's name field can never be `nil`.

Reply:
I thought so too, but then I realized that even if it's impossible, the `sortkey()` shouldn't trigger an error for that situation as it's responsability is to sort and not to validate what it is sorting. That's why I wrote it this way.